client/src/observatory/results.tsx

```diff
@@ -0,0 +1,668 @@
+              <span className="label">Score:</span>
+            </a>{" "}
+            {result.scan.score}/100
+          </div>
+          <div>
+            <a href="#scan_history">
+              <span className="label">Scan Time:</span>
+            </a>{" "}
+            {new Date(result.scan.scanned_at).toLocaleString([], {
+              dateStyle: "medium",
+              timeStyle: "medium",
+            })}
+          </div>
+          <a href="docs/scoring##tests-and-score-modifiers">
+            <span className="label">Tests Passed:</span>
+          </a>{" "}
+          {result.scan.tests_passed}/{result.scan.tests_quantity}
+        </section>
+        <section className="actions">
+          {!isServer && (
+            <CountdownButton
+              host={host}
+              from={result.scan.scanned_at}
+              duration={60}
+              title="Rescan"
+              onClickHandler={rescanTrigger}
+            />
+          )}
+          <div className="scan-another">
+            <InternalLink
+              to="../"
+              onClick={() => gleanClick(`${OBSERVATORY}: click: scan-another`)}
+            >
+              Scan another website
+            </InternalLink>
+          </div>
+        </section>
+      </section>
+    </>
+  );
+}
+
+function hostAsRedirectChain(host, result: ObservatoryResult) {
+  const chain = result.tests.redirection?.route;
+  if (!chain || chain.length < 1) {
+    return host;
+  }
+  try {
+    const firstUrl = new URL(chain[0]);
+    const lastUrl = new URL(chain[chain.length - 1]);
+    if (firstUrl.hostname === lastUrl.hostname) {
+      return host;
+    }
+    return `${firstUrl.hostname} → ${lastUrl.hostname}`;
+  } catch (e) {
+    return host;
+  }
+}
+
+function CountdownButton({
+  host,
+  from,
+  duration,
+  title,
+  onClickHandler,
+}: {
+  host: string;
+  from: string;
+  duration: number;
+  title: string;
+  onClickHandler: Function;
+}) {
+  function calculateRemainingTime() {
+    const endTime = new Date(from).getTime() + duration * 1000;
+    return Math.max(0, endTime - new Date().getTime());
+  }
+  const [remainingTime, setRemainingTime] = useState(calculateRemainingTime());
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      setRemainingTime(calculateRemainingTime());
+    }, 1000);
+
+    return () => clearInterval(interval);
+  });
+
+  const gleanClick = useGleanClick();
+
+  function rescan() {
+    gleanClick(`${OBSERVATORY}: click: rescan`);
+    onClickHandler();
+  }
+
+  const isExpired = remainingTime <= 0;
+  const remainingSecs = Math.floor(remainingTime / 1000) + 1;
+  const progressPercent = (remainingSecs * 100) / 60;
+  return !isExpired ? (
+    <Button isDisabled={true}>
+      <div
+        className="progress"
+        style={{
+          background: `conic-gradient(var(--button-color) 0grad, ${progressPercent}%, rgba(0,0,0,0) ${progressPercent}% 100%)`,
+        }}
+      ></div>
```

Comment:
I'm not sure if this might warrant an aria label?

Reply:
Added `role` and `aria-labelledby`